File: 백준/Silver/11060. 점프 점프/점프 점프.rs
use std::{
    collections::VecDeque,
    io::{self, stdin},
};

fn main() {
    let buf = io::read_to_string(stdin()).unwrap();
    let mut input = buf.split_ascii_whitespace().flat_map(str::parse::<usize>);
    let mut f = || input.next().unwrap();
    let n = f();
    let nums: Vec<usize> = (0..n).map(|_| f()).collect();
    let mut q = VecDeque::new();
    q.push_back((0, 0));
    let mut min_count = n;
    let mut visited = vec![false; n];
    while let Some((idx, count)) = q.pop_front() {
        if idx == n - 1 {
            min_count = min_count.min(count);
            print!("{min_count}");
            return;
        }
        for i in 1..=nums[idx] {
            if idx + i < n && nums[idx + i] > 0 && !visited[idx + i] {
                q.push_back((idx + i, count + 1));
                visited[idx + i] = true;
            }
        }
    }
    print!("-1");
}

File: 백준/Silver/14248. 점프 점프/점프 점프.py
import sys
from collections import deque

input = sys.stdin.readline

n = int(input())
A = list(map(int, input().split()))
s = int(input())

visited = [False] * n
visited[s - 1] = True
dq = deque([s - 1])
while dq:
    pos = dq.popleft()
    for d in (-A[pos], A[pos]):
        next = pos + d
        if 0 <= next < n and not visited[next]:
            visited[next] = True
            dq.append(next)

print(sum(visited))
